fe/fe-core/src/main/java/org/apache/doris/analysis/TimeV2Literal.java

```diff
@@ -0,0 +1,301 @@
+    protected void init(String s) throws AnalysisException {
+        s = normalize(s);
+        // start parse string
+        String[] parts = s.split(":");
+        if (parts.length != 3) {
+            throw new AnalysisException("Invalid format, must have 3 parts separated by ':'");
+        }
+        int scale = 0;
+        try {
+            hour = Float.parseFloat(parts[0]);
+        } catch (NumberFormatException e) {
+            throw new AnalysisException("Invalid hour format", e);
+        }
+
+        try {
+            minute = Long.parseLong(parts[1]);
+        } catch (NumberFormatException e) {
+            throw new AnalysisException("Invalid minute format", e);
+        }
+
+        String[] secondParts = parts[2].split("\\.");
+        if (secondParts.length > 2) {
+            throw new AnalysisException("Invalid second format");
+        }
+
+        try {
+            second = Long.parseLong(secondParts[0]);
+        } catch (NumberFormatException e) {
+            throw new AnalysisException("Invalid second format", e);
+        }
+
+        if (secondParts.length == 2) {
+            String microStr = secondParts[1];
+            scale = microStr.length();
+
+            if (scale > 6) {
+                microStr = microStr.substring(0, 6);
+            }
+
+            StringBuilder sb = new StringBuilder(microStr);
+            while (sb.length() < 6) {
+                sb.append('0');
+            }
+
+            try {
+                microsecond = Long.parseLong(sb.toString());
+            } catch (NumberFormatException e) {
+                throw new AnalysisException("Invalid microsecond format", e);
+            }
+        } else {
+            microsecond = 0;
+            scale = 0;
+        }
+
+        this.type = ScalarType.createTimeV2Type(scale);
+        if (checkRange(hour, minute, second, microsecond)) {
+            throw new AnalysisException("time literal [" + s + "] is out of range");
+        }
+    }
+
+    @Override
+    protected String toSqlImpl() {
+        return "\"" + getStringValue() + "\"";
+    }
+
+    @Override
+    protected void toThrift(TExprNode msg) {
+        msg.node_type = TExprNodeType.TIMEV2_LITERAL;
+        msg.timev2_literal = new TTimeV2Literal(getValue());
+    }
+
+    @Override
+    public boolean isMinValue() {
+        return hour == MIN_TIME.getHour() && minute == 59 && second == 59 && microsecond == 999999;
+    }
+
+    @Override
+    public int compareLiteral(LiteralExpr expr) {
+        return 0;
+    }
+
+    @Override
+    public String getStringValue() {
+        double h = Math.max(Math.min(hour, MAX_TIME.getHour()), MIN_TIME.getHour());
+        long m = Math.max(Math.min(minute, MAX_TIME.getMinute()), MIN_TIME.getMinute());
+        long s = Math.max(Math.min(second, MAX_TIME.getSecond()), MIN_TIME.getSecond());
+        long ms = Math.max(Math.min(getMicroSecond(), MAX_TIME.getMicroSecond()), MIN_TIME.getMicroSecond());
```

Comment:
should not do min max here, u need to ensure every lteral is valid after it be created